### PR TITLE
Create yard

### LINKED ISSUE #22: Host - Create Yard
As a host,
When I visit the host dashboard,
I want to click the button to add a new yard,
So I can be taken to the new yard form and fill out yard information including:

- [x] name
- [x] location
- [x] description
- [x] availability
- [x] photo URLs
- [x]  Checkboxes next tobooking purpose

When I fill out the form and click ‘Create Yard’
Then I am taken back to the host dashboard page
And I see the yard I just created displayed in the list of yards.

Wireframe:
Screen Shot 2021-04-12 at 7 10 53 PM

Subtasks:
1. Yard table, event type, and yard event type tables must be created beforehand.

### PR DESCRIPTION
**Pull request by DougWelchons**
### Add
* EngineService file
    * create_yard method
    * all_purposes method
* figaro, webmock, and ostruct gems
* stubbed create_yard & all_purposes api calls for feature tests

close #22

### CODE REVIEW

File: app/controllers/host/yards_controller.rb [hunk 1]
```diff
@@ -1,5 +1,33 @@
...
 class Host::YardsController < ApplicationController
   def new
-    
+    info = EngineService.all_purposes
+
+    @purposes = info[:data].reduce([]) do |array, obj_info|
+      array << OpenStruct.new(obj_info)
+      array
+    end
+  end
```
Comment (jbranham1) [inline app/controllers/host/yards_controller.rb:10]: Great job! I would like to see this refactored into a facade eventually!

File: app/controllers/host/yards_controller.rb [hunk 2]
```diff
@@ -1,5 +1,33 @@
...
+
+  def create
+    yard_params = {
```
Comment (jbranham1) [inline app/controllers/host/yards_controller.rb:13]: Can we make this a private method to require certain yard params?

**Review by jbranham1 — CHANGES_REQUESTED** [review]
Overall this looks fantastic! I did have a comment about a refactoring opportunity!

File: app/controllers/host/yards_controller.rb (hunk 1 above)
Comment (domo2192) [reply app/controllers/host/yards_controller.rb:10]: I really like how you both did this

**Review by domo2192 — APPROVED** [review]
I think it looks great

File: app/controllers/host/yards_controller.rb (hunk 1 above)
Comment (amsmyth1) [reply app/controllers/host/yards_controller.rb:10]: Yay, I love openstruct!

File: app/controllers/host/yards_controller.rb (hunk 2 above)
Comment (amsmyth1) [reply app/controllers/host/yards_controller.rb:13]: I took care of this and will merge the branch.

Comment (amsmyth1) [reply app/controllers/host/yards_controller.rb:13]: Jordan and I will review this!

**Comment by amsmyth1** [comment]
Everything looks great. I pulled down to my local and created the environmental variable and all tests are passing!

**Review by amsmyth1 — APPROVED** [review]
Everything looks good. Jordan and I will build from here and review JBs comment regarding making yard params a private method.

**Review by jbranham1 — APPROVED** [review]

**Pull request merged**